logendpoint.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * This file is part of the MAVLink Router project
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "logendpoint.h"
+
+#include <dirent.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "log.h"
+#include "mainloop.h"
+#include "util.h"
+
+#define MAX_LOG_FILE_PREFIX 999
+
+static bool _logging_start_timeout_cb(void *data)
+{
+    LogEndpoint *log = static_cast<LogEndpoint *>(data);
+    return log->_start_timeout();
+}
+
+static bool _alive_timeout_cb(void *data)
+{
+    LogEndpoint *log = static_cast<LogEndpoint *>(data);
+    return log->_alive_timeout();
+}
+
+void LogEndpoint::_send_msg(const mavlink_message_t *msg, int target_sysid)
+{
+    uint8_t data[MAVLINK_MAX_PACKET_LEN];
+    struct buffer buffer {
+        0, data
+    };
+
+    buffer.len = mavlink_msg_to_send_buffer(data, msg);
+    _mainloop->route_msg(&buffer, target_sysid, msg->sysid);
+
+    _stat.read.total++;
+    _stat.read.handled++;
+    _stat.read.handled_bytes += msg->len;
+    if (msg->magic == MAVLINK_STX)
```

Comment:
mavlink1 vs mavlink2 is seemless if you use mavlink library functions.

The intention was not to use it in most of the places, and use the wire format directly on the core router part. Then this ended up being spread across different places, which is in need of a cleanup patch.

@edersondisouza in this particular case, why are you not using `buffer.len` since you only want the size of the msg including header, which should be already in this variable.